mllib/src/main/scala/org/apache/spark/mllib/classification/NaiveBayes.scala

```diff
@@ -156,9 +191,12 @@ object NaiveBayesModel extends Loader[NaiveBayesModel] {
  * document classification.  By making every vector a 0-1 vector, it can also be used as
  * Bernoulli NB ([[http://tinyurl.com/p7c96j6]]). The input feature values must be nonnegative.
  */
-class NaiveBayes private (private var lambda: Double) extends Serializable with Logging {
+class NaiveBayes private (private var lambda: Double,
+                          var modelType: NaiveBayesModels) extends Serializable with Logging {
```

Comment:
I'd recommend making the var private and providing a getModelType method.  That limits the public API more so that it will be easier to change if we ever need to deprecate or modifyf the "NaiveBayesModels" type.